lib/Mojo/WebSocketProxy/Backend/JobAsync.pm

```diff
@@ -106,56 +88,79 @@ Implements the L<Mojo::WebSocketProxy::Backend/call_rpc> interface.
 
 sub call_rpc {
     my ($self, $c, $req_storage) = @_;
-    my $method   = $req_storage->{method};
+    my $method = $req_storage->{method};
     my $msg_type = $req_storage->{msg_type} ||= $req_storage->{method};
-    $self->{client}->start->get;
 
-    $req_storage->{call_params} ||= {};
+    # We'd like to provide some flexibility for people trying to integrate this into
+    # other systems, so any combination of Job::Async::Client, Job::Async and/or IO::Async::Loop
+    # instance can be provided here.
+    $self->{client} //= do {
+        # We don't hold a ref to this, since that might introduce unfortunate cycles
+        my $loop //= do {
```

Comment:
@nael-binary We are getting `perlcritic` warnings about this line. Isn't it better to be rewritten as `$self->{loop} = `?

Reply:
yes actually here its better to use `$self->{loop}`